org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -226,7 +226,8 @@ public static CompletionParams toCompletionParams(URI fileUri, int offset, IDocu
 		Position start = toPosition(offset, document);
 		final var param = new CompletionParams();
 		try {
-			String positionCharacter = document.get(offset-1, 1);
+			int positionCharacterOffset = offset > 0 ? offset-1 : offset;
+			String positionCharacter = document.get(positionCharacterOffset, 1);
```

Comment:
It's most likely failing if document is empty.

Reply:
The `BadLocationException` in the stack trace has been triggered by LSPEclipseUtils.java:229. The document.get method fails if the offset is <0. 
What is the expected behavior for `org.eclipse.lsp4e.LSPEclipseUtils.toCompletionParams(URI, int, IDocument, char[])` in this case? I assumed 0 would be approriate for the offset value.

Reply:
Which test case fails here?